server/update-workflow-interface/update-workflow-interface.md

```diff
@@ -0,0 +1,217 @@
+# Proposal: Update Workflow Server Interface
+
+Currently (as of version 1.15), ad-hoc runtime interaction with Temporal
+workflows is achieved through queries and signals. The former technique can
+interrogate the state of a workflow and returns a result but must not _change_
+the state of the workflow, while the latter can change workflow state but does
+not natively support return values [^1]. 
+
+The missing piece in runtime workflow interaction is an affordance that
+provides both (1) the ability to change workflow state, and (2) a mechanism to
+observe the result of the same without polling or setting up callback
+structures. This feature will be workflow updates.
+
+This proposal describes the server API (i.e. gRPC) for such a feature. We leave
+implementation details for a deeper design review and instead relate the API,
+its expected usage and the guarantees provided.
+
+## Concepts
+
+### Update
+
+An Update is the execution of arbitrary workflow logic (subject to the standard
+workflow code constraints on determinism) triggered by an UpdateRequest and
+eventually resulting in a single UpdateResponse. The processing of an Update is
+performed by workflow code and as such the timeliness of the UpdateResponse is
+dependent on worker resources being available to process the Update. A given
+Update applies to exactly one workflow execution. The lifetime of an updated
+begins at the first UpdateRequest triggering said Update and ends when the
+UpdateResponse has been made durable. Consequently, the lifetime of an Update
+may exceed the lifetime of the RPC request/response that initiates it.
+
+### Update ID
+
+The unique identifier of an Update. Generated by the Temporal server.
+
+### UpdateRequest
+
+An UpdateRequest is a remote call originating either from outside the Temporal
+system entirely or from another Temporal workflow.  UpdateRequests are
+deduplicated such that issuance of the same UpdateRequest multiple times is
+effectively the same as issuing a single request. The UpdateRequest includes an
+indication of the workflow execution at which the Update is targeted.
+UpdateRequests are issued over gRPC or through SDK-specific APIs wrapping the
+same. Both issuing and receiving UpdateRequests becomes a durable part of a
```

Comment:
It's not exactly clear what does it look like. Can we add a piece of history? Will it be the same with the Signal - UpdateRequest event followed by WFT?